File: ticket-backend/src/main/java/aics/domain/provider/dtos/ProviderDto.java
package aics.domain.provider.dtos;

import aics.domain.hall.dtos.HallListItemDto;
import aics.domain.provider.entities.Provider;
import lombok.Data;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.util.List;
import java.util.stream.Collectors;

@Data
@Accessors(chain = true)
public class ProviderDto implements Serializable {
    private Long providerId;
    private String name;
    private String address;
    private String phone;
    private String description;
    private String googleMapsSrc;
    private List<HallListItemDto> halls;

    public static ProviderDto fromProvider(Provider provider) {
        if (provider == null) {
            return null;
        }
        return new ProviderDto()
            .setProviderId(provider.getProviderId())
            .setName(provider.getName())
            .setAddress(provider.getAddress())
            .setPhone(provider.getPhone())
            .setDescription(provider.getDescription())
            .setGoogleMapsSrc(provider.getGoogleMapsSrc())
            .setHalls(provider.getHalls().stream().map(HallListItemDto::fromHall).collect(Collectors.toList()));
    }
}

File: ticket-frontend/src/modules/provider/provider-dto.ts
import { TypeUtils } from '../core/type-utils';
import { HallListItemDto } from '../hall/hall-list-item-dto';

export class ProviderDto {
    providerId: number | null = null;
    name: string = '';
    address: string = '';
    phone: string = '';
    description: string = '';
    googleMapsSrc: string = '';
    halls: HallListItemDto[] = [];

    static fromObj(obj: any): ProviderDto | null {
        if (!obj) {
            return null;
        }
        const providerDto: ProviderDto = new ProviderDto();
        providerDto.providerId = obj.providerId;
        providerDto.name = obj.name;
        providerDto.address = obj.address;
        providerDto.phone = obj.phone;
        providerDto.description = obj.description;
        providerDto.googleMapsSrc = obj.googleMapsSrc;
        providerDto.halls = HallListItemDto.listFromObjList(obj.halls);
        return providerDto;
    }

    static listFromObjList(objs: any[]): ProviderDto[] {
        if (!objs) {
            return [];
        }
        return objs.map(ProviderDto.fromObj).filter(TypeUtils.isNonNullable);
    }
}